### PR TITLE
Add mid-circuit measurement before conditional operation in random_circuit

### LINKED ISSUE #9016: `random_circuit` does not include mid-circuit measurements
### What should we add?

`qiskit.circuit.utils.random_circuit` has a keyword argument `measure=False` that allows opt-in inclusion of an end-of-circuit measurement of all qubits: https://github.com/Qiskit/qiskit-terra/blob/ad7e609d10b02fd98090b0fac132113f329825b6/qiskit/circuit/random/utils.py#L50

It would be good to also be able to generalize this to be able to generate circuits which can additionally (and optionally) make of use mid-circuit measurement.

### PR DESCRIPTION
**Pull request by MaldoAlberto**
Fixes #9016

### Summary

New version of the PR https://github.com/Qiskit/qiskit-terra/pull/9207

### CODE REVIEW

File: releasenotes/notes/fix_9016-2e8bc2cb10b5e204.yaml
Comment (Eric-Arellano) [inline releasenotes/notes/fix_9016-2e8bc2cb10b5e204.yaml:5]:
```suggestion
    ``qiskit.circuit.random.random_circuit``, the conditional operations will
```

**Review by Eric-Arellano — APPROVED** [review]
Lgtm Python code quality-wise. But I don't understand the underlying functionality, so needs review from another maintainer.

**Review by kdk — APPROVED** [review]
Thanks for picking this up. This LGTM!

**Pull request merged**
